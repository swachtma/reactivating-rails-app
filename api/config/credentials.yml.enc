NruSqAfDHjCumPIyJn6xsY9LOZuE3ZzX2o/1QTdVkugVslYlIp4h/gmrnNPKDUKYZ8vGEGHPJ5EwlOSClaonGjgUB4yUG4tz8I0T++XLbiEU8qhxATlTf+s9JNfusXdgIB2ZFO4ABSDKhxhzPLbHYJVvLghx3WmwMMpO+GBgzlp94DLCX2IR38jeHEydFPy73oteodKVbTBp0g91v8DhpnZUTWHYb0JdqS+otrdrS1mar9MlcMnTpb8chvLisrrvE0SR/QUagAbd5UD/d02NczFYmAdKsyztbryVyakhgosZhCJFY3KNwxayxDC3sF5xFQHgli8qJNXNAhzaYDVf+n7D9k6g9BHTV5YRXdICgYORVWo+a9EhjBTmVwQsJN+IrdZaqxQVoa2DMEA3CVjOlp1wtLNwHbHArmId--8+ArDlMZvB3FEYTs--m4O75Mk62PJos0rR8zw91Q==