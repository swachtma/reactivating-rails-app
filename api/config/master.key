dc4eeb306d00d2a5e6993b94e449787e