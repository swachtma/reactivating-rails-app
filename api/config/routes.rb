Rails.application.routes.draw do
  scope :api do
    get 'github/', to: "authentication#github", format: false
    get 'hydrate_user/', to: "authentication#show"
    get 'nodes/', to: "nodes#index"
    get 'chapters/', to: "chapters#index"
    resources :bookmarks, only: [:index, :create]
  end
end