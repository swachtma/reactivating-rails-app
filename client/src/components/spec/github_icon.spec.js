/* global expect, jest */
import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

import config from '../../test_helper';
import GithubIcon from '../github_icon';

let test_props = {
  location: {
    pathname: '/chapter/15',
    type: 'CHAPTER_ROUTE',
    payload: {
      chapter_id: 15
    },
    prev: {
      pathname: '',
      type: '',
      payload: {}
    },
    kind: 'load',
    routesMap: {
      HOME_ROUTE: '/',
      CHAPTER_ROUTE: '/chapter/:chapter_id',
      AUTH_ROUTE: '/auth/:token/:bounce_path',
      ERROR_ROUTE: '/error'
    }
  }
};

let w = shallow(<GithubIcon {...test_props} />);

describe("<GithubIcon /> renders",()=>{
  it("according to snapshot", () => expect(w).toMatchSnapshot());
});