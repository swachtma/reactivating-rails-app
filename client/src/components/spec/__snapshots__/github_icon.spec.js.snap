// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GithubIcon /> renders according to snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GithubIcon
    location={
        Object {
            "kind": "load",
            "pathname": "/chapter/15",
            "payload": Object {
              "chapter_id": 15,
            },
            "prev": Object {
              "pathname": "",
              "payload": Object {},
              "type": "",
            },
            "routesMap": Object {
              "AUTH_ROUTE": "/auth/:token/:bounce_path",
              "CHAPTER_ROUTE": "/chapter/:chapter_id",
              "ERROR_ROUTE": "/error",
              "HOME_ROUTE": "/",
            },
            "type": "CHAPTER_ROUTE",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Icon
        as="i"
        name="github"
        size="big"
        style={
                Object {
                        "marginRight": 0,
                      }
        }
/>,
      "fitted": true,
      "href": "https://github.com/login/oauth/authorize?client_id=a86bc65853ae65d3be52&redirect_uri=http%3A%2F%2Flocalhost%3A3000%2Fapi%2Fgithub%2F%3Fbounce_path%3D%7B%22type%22%3A%22CHAPTER_ROUTE%22%2C%22payload%22%3A%7B%22chapter_id%22%3A15%7D%7D",
      "name": "Sign in with GitHub",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "as": "i",
        "name": "github",
        "size": "big",
        "style": Object {
          "marginRight": 0,
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Icon
          as="i"
          name="github"
          size="big"
          style={
                    Object {
                              "marginRight": 0,
                            }
          }
/>,
        "fitted": true,
        "href": "https://github.com/login/oauth/authorize?client_id=a86bc65853ae65d3be52&redirect_uri=http%3A%2F%2Flocalhost%3A3000%2Fapi%2Fgithub%2F%3Fbounce_path%3D%7B%22type%22%3A%22CHAPTER_ROUTE%22%2C%22payload%22%3A%7B%22chapter_id%22%3A15%7D%7D",
        "name": "Sign in with GitHub",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "as": "i",
          "name": "github",
          "size": "big",
          "style": Object {
            "marginRight": 0,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
